clicommand/agent_start.go

```diff
@@ -761,6 +762,17 @@ var AgentStartCommand = cli.Command{
 			l.Fatal("The given tracing backend %q is not supported. Valid backends are: %q", cfg.TracingBackend, maps.Keys(tracetools.ValidTracingBackends))
 		}
 
+		if experiments.IsEnabled("leader-api") {
+			// Try to be the leader - no worries if not.
+			leaderSvr, err := leaderapi.NewServer(leaderapi.LeaderSocketPath)
+			if err != nil {
+				l.Warn("Couldn't become leader: %v", err)
+			}
+			if leaderSvr != nil {
+				defer leaderSvr.Shutdown(ctx)
+			}
+		}
+
```

Comment:
What happens when a leader dies? It looks like nothing will be listening on the socket and the clients will all fail. That's not too bad, but I think some kind of promotion procedure would be useful.

Reply:
Also a good idea to print an info log when this does become the leader.

Reply:
Done